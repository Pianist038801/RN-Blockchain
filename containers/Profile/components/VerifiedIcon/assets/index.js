import logoGreen from './logo-green-white.png'
import logoYellow from './logo-yellow-white.png'
import logoGrey from './logo-grey-white.png'

export default {
  logoGreen,
  logoYellow,
  logoGrey
}
