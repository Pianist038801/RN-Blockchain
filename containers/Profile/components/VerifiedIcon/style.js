import { Dimensions } from 'react-native'

const { width } = Dimensions.get('window')

const style = {
  profileEntityVerification: {
    width: width * 0.1,
    height: width * 0.1,
    resizeMode: 'contain'
  }
}

export default style
