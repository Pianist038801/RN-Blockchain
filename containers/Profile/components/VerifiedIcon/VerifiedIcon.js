import React from 'react'
import PropTypes from 'prop-types'
import { Image, TouchableOpacity } from 'react-native'
import ActionSheet from 'react-native-actionsheet-native'
import BaseComp from '../../../../components/base_component.js'
import Assets from './assets'
import style from './style'

class VerifiedIcon extends BaseComp {
  verifyContext = () => {
    const BUTTONS = [
      'Request verification',
      'Cancel'
    ]

    ActionSheet.showActionSheetWithOptions({
      options: BUTTONS,
      cancelButtonIndex: BUTTONS.length - 1,
    }, (idx) => {
      if (idx === 0) {
        const { entity, type } = this.props
        let searchField;
        switch (type) {
          case 'job':
            searchField = entity.company
            break;
          case 'education':
            searchField = entity.institution
            break;
          case 'award':
            searchField = entity.organisation
            break;
        }

        this.context.store.dispatch({
          type: 'START_VERIFICATION_REQUEST',
          payload: {
            searchField,
            type,
            id: entity.id
          }
        })
      }
    })
  }

  getEntityStatus(entity) {
    if (!entity.pending_verification && !entity.verified) {
      return 0
    }

    if (entity.verified) {
      return 2
    }

    if (entity.pending_verification && !entity.verified) {
      return 1
    }

    return 0
  }

  render() {
    const entity = this.props.entity
    const status = this.getEntityStatus(entity)
    let source
    let requestable = false

    switch (status) {
      case 1:
        source = Assets.logoYellow
        break
      case 2:
        source = Assets.logoGreen
        break
      default:
        requestable = true
        source = Assets.logoGrey
    }

    const image = (
      <Image
        style={style.profileEntityVerification}
        source={source}
      />
    )

    if (requestable) {
      return (
        <TouchableOpacity
          onPress={this.verifyContext}
        >
          {image}
        </TouchableOpacity>
      )
    }

    return image
  }
}

VerifiedIcon.propTypes = {
  type: PropTypes.string,
  entity: PropTypes.shape({
    pending_verification: PropTypes.bool,
    verified: PropTypes.bool
  })
}

export default VerifiedIcon
