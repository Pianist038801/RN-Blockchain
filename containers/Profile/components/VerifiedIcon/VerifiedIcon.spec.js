import React from 'react'
import renderer from 'react-test-renderer'
import VerifiedIcon from './index'

describe('containers/Profile/components/VerifiedIcon', () => {
  describe('when pending_verification and verified are false', () => {
    it('renders correctly', () => {
      const props = {
        entity: {
          id: 1,
          pending_verification: false,
          verified: false
        }
      }

      const tree = renderer.create(
        <VerifiedIcon {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when verified is true', () => {
    it('renders correctly', () => {
      const props = {
        entity: {
          id: 1,
          verified: true
        }
      }

      const tree = renderer.create(
        <VerifiedIcon {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when pending_verification is true and verified is false', () => {
    it('renders correctly', () => {
      const props = {
        entity: {
          id: 1,
          pending_verification: true,
          verified: false
        }
      }

      const tree = renderer.create(
        <VerifiedIcon {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
