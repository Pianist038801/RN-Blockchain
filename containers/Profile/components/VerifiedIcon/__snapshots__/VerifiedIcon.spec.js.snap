// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/VerifiedIcon when pending_verification and verified are false renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={1}
    style={
      Object {
        "height": 75,
        "resizeMode": "contain",
        "width": 75,
      }
    }
  />
</View>
`;

exports[`containers/Profile/components/VerifiedIcon when pending_verification is true and verified is false renders correctly 1`] = `
<Image
  source={1}
  style={
    Object {
      "height": 75,
      "resizeMode": "contain",
      "width": 75,
    }
  }
/>
`;

exports[`containers/Profile/components/VerifiedIcon when verified is true renders correctly 1`] = `
<Image
  source={1}
  style={
    Object {
      "height": 75,
      "resizeMode": "contain",
      "width": 75,
    }
  }
/>
`;
